src/util/crypto_node.ts

```diff
@@ -0,0 +1,49 @@
+import * as crypto from 'crypto'
+import * as assert from 'assert'
+
+const HASH_ALGORITHM = 'sha256'
+const ENCRYPTION_ALGORITHM = 'aes-256-gcm'
+const IV_LENGTH = 12
+const AUTH_TAG_LENGTH = 16
+
+export const randomBytes = crypto.randomBytes
+
+export async function hash (preimage: Buffer): Promise<Buffer> {
+  const h = crypto.createHash(HASH_ALGORITHM)
+  h.update(preimage)
+  return Promise.resolve(h.digest())
+}
+
+export async function encrypt (pskEncryptionKey: Buffer, ...buffers: Buffer[]): Promise<Buffer> {
+  const iv = crypto.randomBytes(IV_LENGTH)
+  const cipher = crypto.createCipheriv(ENCRYPTION_ALGORITHM, pskEncryptionKey, iv)
+
+  const ciphertext = []
+  for (let buffer of buffers) {
+    ciphertext.push(cipher.update(buffer))
+  }
+  ciphertext.push(cipher.final())
+  const tag = cipher.getAuthTag()
+  ciphertext.unshift(iv, tag)
+  return Promise.resolve(Buffer.concat(ciphertext))
```

Comment:
Wonder why node itself doesn't offer an async api for crypto operations ?

Reply:
Not sure; maybe the overhead would negate the benefits.